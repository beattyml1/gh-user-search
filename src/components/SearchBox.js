// props: { onSearch: function, label, onValueChange }
import * as React from "react";
export function SearchBox(props) {
    let id = props.id || "search";
    let onValueChange = props.onValueChange || (() => {});

    return (
        <form role={"search"} onSubmit={props.onSearch}>
            <input type={"search"} role={"search"} placeholder={props.label} aria-label={props.label}
                   onChange={props.onValueChange} autoFocus={true} spellCheck="false"/>
            <button type={"submit"}>Search</button>
        </form>
    );
}