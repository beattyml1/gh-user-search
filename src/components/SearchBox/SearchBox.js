// props: { onSearch: function, label, onValueChange }
import * as React from "react";
import './SearchBox.css';

export class SearchBox extends React.Component {
    constructor() {
        super();
        this.state = { query: '' }
    }
    onValueChange = e => {
        this.setState({query: e.target.value});
        if (this.props.onValueChange) this.props.onValueChange(e.target.value);
    };
    onSearch = e => {
        this.props.onSearch(this.state.query);
        e.preventDefault();
        return false;
    }

    render = () => {
        return (
            <form role={"search"} onSubmit={this.onSearch} className={"search"}>
                <input type={"search"} role={"search"} name={"query"} placeholder={this.props.label} value={this.state.query||''}
                       aria-label={this.props.label} className={"search-box"} onChange={this.onValueChange}
                       autoFocus={true} spellCheck="false"/>
                <button type={"submit"} className={"search-submit"}>Search</button>
            </form>
        );
    }
}