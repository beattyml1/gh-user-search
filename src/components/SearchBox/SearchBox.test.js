import * as React from 'react';
import ReactDOM from 'react-dom';
import {SearchBox} from './SearchBox';
import renderer from 'react-test-renderer';

it('creates expected mark up for 0 results', () => {
    const tree = renderer.create(
        <SearchBox label={"Search Something"}/>
    ).toJSON();
    expect(tree).toMatchSnapshot()
});