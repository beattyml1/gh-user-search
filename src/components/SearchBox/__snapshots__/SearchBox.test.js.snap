// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creates expected mark up for 0 results 1`] = `
<form
  className="search"
  onSubmit={[Function]}
  role="search"
>
  <input
    aria-label="Search Something"
    autoFocus={true}
    className="search-box"
    name="query"
    onChange={[Function]}
    placeholder="Search Something"
    role="search"
    spellCheck="false"
    type="search"
    value=""
  />
  <button
    className="search-submit"
    type="submit"
  >
    Search
  </button>
</form>
`;
